### PR TITLE
[SDRS] New `resource/opentelekomcloud_sdrs_protected_instance_v1`

### LINKED ISSUE #942: [SDRS] protected instance config missing?
Hi there,
sorry if i am simply blind, but i can't find the possibility to configure a protected instance to be protected by an sdrs protection group. How may one protect an instance in this way?

Thanks in advance!

# Terraform Version
```
Terraform v0.12.29
+ provider.local v2.0.0
+ provider.opentelekomcloud v1.21.2
+ provider.template v2.2.0
```

# Affected Resource(s)
- opentelekomcloud_sdrs_protectiongroup_v1

### PR DESCRIPTION
**Pull request by lego963**
## Summary of the Pull Request
Add new `resource/opentelekomcloud_sdrs_protected_instance_v1`
Resolves: #942

## PR Checklist

* [x] Refers to: #942
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
=== RUN   TestAccSdrsProtectedInstanceV1_basic
--- PASS: TestAccSdrsProtectedInstanceV1_basic (300.06s)
PASS

Process finished with exit code 0

=== RUN   TestAccSdrsProtectedInstanceV1_importBasic
--- PASS: TestAccSdrsProtectedInstanceV1_importBasic (279.67s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

**Review by anton-sidelnikov — DISMISSED** [review]

File: docs/resources/sdrs_protected_instance_v1.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+* `name` - (Required) The name of a protected instance.
+
+* `description` - (Optional) The description of a protected instance. Changing this creates a new instance.
+
+* `group_id` - (Required) Specifies the ID of the protection group where a protected instance is added.
+
+* `server_id` - (Required) Specifies the ID of the production site server.
+
+-> When the API is successfully invoked, the DR site server will be automatically created.
```
Comment (outcatcher) [inline docs/resources/sdrs_protected_instance_v1.md]: What is DR site server?

File: docs/resources/sdrs_protected_instance_v1.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+The following arguments are supported:
+
+* `name` - (Required) The name of a protected instance.
+
+* `description` - (Optional) The description of a protected instance. Changing this creates a new instance.
+
+* `group_id` - (Required) Specifies the ID of the protection group where a protected instance is added.
+
+* `server_id` - (Required) Specifies the ID of the production site server.
```
Comment (outcatcher) [inline docs/resources/sdrs_protected_instance_v1.md]: Does this mean ID of the protected compute instance?

File: docs/resources/sdrs_protected_instance_v1.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+-> When the API is successfully invoked, the DR site server will be automatically created.
+
+* `subnet_id` - (Optional) Specifies the network ID of the subnet for the primary NIC on the DR site server.
+  The value is the same as that of `network_id` obtained using the VPC API.
```
Comment (outcatcher) [inline docs/resources/sdrs_protected_instance_v1.md]: >The value is the same as that of `network_id` obtained using the VPC API.

For the VPC API it is just a `subnet.id`

**Review by outcatcher — CHANGES_REQUESTED** [review]

File: docs/resources/sdrs_protected_instance_v1.md (hunk 2 above)
Comment (lego963) [reply docs/resources/sdrs_protected_instance_v1.md]: yes

File: docs/resources/sdrs_protected_instance_v1.md (hunk 1 above)
Comment (lego963) [reply docs/resources/sdrs_protected_instance_v1.md]: Disaster Recovery server, new server

File: docs/resources/sdrs_protected_instance_v1.md (hunk 2 above)
Comment (outcatcher) [reply docs/resources/sdrs_protected_instance_v1.md]: can you mention that?

**Review by outcatcher — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Comment by lego963** [comment]
regate

**Pull request merged**